### PR TITLE
Extended README

### PR DESCRIPTION
**Pull request by Damtev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,102 @@
...
+- **Disequality constraints** &ndash; <code>(&alpha; &ne; &beta;)</code> asserts that two logic terms &alpha; and &beta; are not equal and can 
+never be made equal through unification.
+- **Typed logic terms** &ndash; strong static typing disallows unification of logic terms with incompatible types
+in compile time.
+- **[Wildcard logic variables](https://drive.google.com/file/d/1RdtwC2kmzHK7Sz3fO_Hq9AgMOxwr5m-2/view)** &ndash; 
+extending the expressive power of miniKanren with a limited form of universal quantification.
+
+# Examples
+
+Let's see how generating all permutations of some Peano number could be implemented.
```
Comment (Kakadu) [inline README.md:35]: list is forgotten

File: README.md [hunk 2]
```diff
@@ -1,2 +1,102 @@
...
-An implementation of miniKanren in Kotlin
+# What is klogic?
+
+klogic is a strong-typed implementation of relational programming language [miniKanren](http://minikanren.org/) into Kotlin. 
+It is largely inspired by [OCanren](https://github.com/PLTools/OCanren) &ndash; a widely known embedding of miniKanren in 
+functional programming language OCaml.
+
+# How can I use it?
+
+Relational programming languages are designed based on the logical programming paradigm and perfectly suite
```
Comment (Kakadu) [inline README.md:9]: are designed based --- коряво

File: README.md [hunk 3]
```diff
@@ -1,2 +1,102 @@
...
+for solving problems that can be expressed as logical queries &ndash; for example, theorem-proving tasks,
+puzzle-solving, etc.
+
+Also, relational languages in general and klogic in particular are well-suited for solving constraint 
+satisfaction problems. Constraint satisfaction problems involve finding a solution that satisfies a set 
+of constraints or conditions. These constraints can be mathematical equations, logical constraints, or
```
[the remark this answers is not in the document]
Comment (Damtev) [reply README.md:15]: Why not? There could be some constraints about elements and sets, not only numeric equations (that are not simply expressed, yes).

File: README.md (hunk 2 above)
Comment (Damtev) [reply README.md:9]: Well, Grammarly does not find the issue

**Pull request merged**